src/Servers/HttpSys/src/AsyncAcceptContext.cs

```diff
@@ -167,6 +167,7 @@ private void AllocateNativeRequest(uint? size = null, ulong requestId = 0)
         if (_overlapped != null)
         {
             boundHandle.FreeNativeOverlapped(_overlapped);
+            _overlapped = null;
         }
 
         _requestContext = _requestContextFactory.CreateRequestContext(size, requestId);
```

Comment:
So the theory is that this or the next line is throwing before _overlapped gets overridden? I agree that CreateRequestContext doesn't seem like something that's super likely to throw. Maybe we could force it by using a test HttpSysListener.MemoryPool that throws from Rent(). 

It still doesn't make sense to me that throwing from AllocateNativeRequest is the cause of the issue though. The customer observed the ArgumentException being thrown from this method, so why wouldn't the customer observe the first exception that lead to the later ArgumentException being thrown from the same method?

Any non-shutdown exception would lead to an error-level "Failed to accept a request." log, right? Did the partner reporting this see anything like this @Tratcher? And did we verify that double-freeing the overlapped handle causes a ArgumentException with the "'overlapped' was not allocated by this ThreadPoolBoundHandle instance." message?

AllocateNativeOverlapped throwing an OOM might be more plausible than CreateRequestContext throwing, but we could throw from Rent() to verify that throwing from one of these lines could lead to just the ArgumentException being observed without the original exception being observed for some reason.